src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -47,6 +47,14 @@ public List<CustomNodeWorkspaceModel> CustomNodesWorkspaces
             private set;
         }
 
+        public bool HasAuthenticationProvider
+        {
+            get
+            {
+                return authenticationProvider != null;
+            }
+        }
```

Comment:
A couple naming related things.  One of which I missed before:

1. Rename to `HasAuthProvider` - this keeps it consistent with the name of the associated type. I am fastidious about such things if you haven't noticed.  :)
2. Let's rename `CustomNodesWorkspaces` to `CustomNodeWorkspaces` - in english, the plural on the adjective of a plural noun is redundant.